### PR TITLE
Allow tenants with NilEndpoints to be deleted

### LINKED ISSUE #1351: Cleanup account job to ignore NilSolrEndpoint error
Background:

When attempting to create a tenant last week, Heather ran into an error where Solr was unavailable because it could not talk to Zookeeper. (https://app.honeybadger.io/projects/48327/faults/34032248). The Solr connection issues have been cleared up, but now we're in a state where the app thinks a tenant exists, but no solr index for that tenant was ever created. Attempting to delete the tenant (from the UI) results in https://app.honeybadger.io/projects/48327/faults/34103318.

Justin's analysis:

cleanup_account_job should be changed to ignore NilSolrEndpoint

this line: https://github.com/samvera-labs/hyku/blob/master/app/jobs/cleanup_account_job.rb#L40

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #1351

This is an alternative to #1387  What do you think @tdonohue?

### CODE REVIEW

File: app/jobs/cleanup_account_job.rb [hunk 1]
```diff
@@ -5,54 +5,28 @@ def perform(account)
...
-    end
-
-    def redis_namespace(account)
-      # Ensure Redis is switched to account's namespace
-      account.redis_endpoint.switch!
-      # Return a Redis::Namespace which is already limited to account's namespace
-      Hyrax::RedisEventStore.instance
+    def cleanup_database(account)
+      Apartment::Tenant.drop(account.tenant)
+    rescue
```
Comment (cbeer) [inline app/jobs/cleanup_account_job.rb]: I'm not thrilled with rescuing everything, but on the other hand, what are we going to do?

**Comment by cbeer** [comment]
👍 This looks like an idiomatic usage of the null object pattern.

**Comment by tdonohue** [comment]
@jcoyne : This approach looks good overall & makes sense.  Thanks for the alternative PR. 

That said, this morning I've stumbled on a few minor bugs in this PR (during testing).  I'll add a commit to your PR branch to fix them.

**Comment by jcoyne** [comment]
Thanks @tdonohue

**Pull request merged**